From 42e2ff146a87819e6eb2eebd8f5dad8bc9c79a55 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 28 Nov 2023 11:36:09 +0100
Subject: [PATCH 426/841] fix(stm32mp2): remove unnecessary assert on
 GPIO_BANK_A value

Remove assert for unexpected value of the define GPIO_BANK_A.

This check is not required as GPIO_BANK_A = 0, the check can be limited
to have GPIO_BANK_A (=0) <= bank <= GPIO_BANK_K with bank = unsigned int.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 7ec1f109727111d9354fe803350bd46594be74ac fix(stm32mp2): remove unnecessary assert on GPIO_BANK_A value]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: If29f69474a6063c8bdf1a627eda63d5928ee6bc0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/342381
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_private.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index f1b363a9d..9f6829070 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -122,7 +122,7 @@ uintptr_t stm32_get_gpio_bank_base(unsigned int bank)
 		return GPIOZ_BASE;
 	}
 
-	assert(GPIO_BANK_A == 0 && bank <= GPIO_BANK_K);
+	assert(bank <= GPIO_BANK_K);
 
 	return GPIOA_BASE + (bank * GPIO_BANK_OFFSET);
 }
@@ -133,7 +133,7 @@ uint32_t stm32_get_gpio_bank_offset(unsigned int bank)
 		return 0;
 	}
 
-	assert(GPIO_BANK_A == 0 && bank <= GPIO_BANK_K);
+	assert(bank <= GPIO_BANK_K);
 
 	return bank * GPIO_BANK_OFFSET;
 }
@@ -144,7 +144,7 @@ unsigned long stm32_get_gpio_bank_clock(unsigned int bank)
 		return CK_BUS_GPIOZ;
 	}
 
-	assert(GPIO_BANK_A == 0 && bank <= GPIO_BANK_K);
+	assert(bank <= GPIO_BANK_K);
 
 	return CK_BUS_GPIOA + (bank - GPIO_BANK_A);
 }
-- 
2.47.2

